src/main/java/core/basesyntax/dao/FruitDaoImpl.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+
+public class FruitDaoImpl implements FruitDao {
+
+    @Override
+    public Integer get(Fruit fruit) {
+        return Storage.storage.getOrDefault(fruit, 0);
+    }
+
+    @Override
+    public Integer update(Fruit fruit, Integer amountToAdd) {
+        if (amountToAdd < 0 && Math.abs(amountToAdd) > Storage.storage.get(fruit)) {

```

Comment:
у тебя тут написано, если количество меньше нуля И, если количество больше чем фруктов в базе, это правильно ? 
собственно поэтому у тебя не вылетает exception, когда ты в тесте вызываешь  update на пустой базе
```suggestion
        if (amountToAdd < 0 || Math.abs(amountToAdd) > Storage.storage.get(fruit)) {

```

Reply:
здесь я проверяю условие когда я продаю количество фруктов больше чем у меня есть. amountToAdd у меня приходит с OperationHandler отрицательное, и я проверяю что на складе больше чем продать нужно, по этому модуль беру. если поставлю | | то любое количество, которое я ложу на склад (если оно больше текущего остатка) будет давать Exception